src/test/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProviderTest.java

```diff
@@ -528,6 +552,53 @@ public void shouldReturnCorrectJob() {
         Assertions.assertEquals(expectedFilteredJob, result);
     }
 
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(longs = {SOME_WRONG_JOB_ID})
+    public void shouldThrowWhenPassWrongRequestToDeleteJob(final Long jobId) {
+        final DeleteJobFilter deleteJobFilter = new DeleteJobFilter(false, jobId, null);
+        Assertions.assertThrows(GridEngineException.class, () -> slurmJobProvider.deleteJob(deleteJobFilter));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideValidNameCasesForRequestsToDeleteJob")
+    public void shouldReturnCorrectUserNameAndJobIdWhenDeleteJob(final String expectedUserName,
+                                                                 final String passedUserName,
+                                                                 final String... commandArgs) {
+        final CommandResult whoamiCommandResult =
+                new CommandResult(Collections.singletonList(SLURM_USER), 0, EMPTY_LIST);
+        mockCommandCompilation(WHOAMI_COMMAND, whoamiCommandResult, WHOAMI_COMMAND);
+
+        final CommandResult scancelCommandResult = new CommandResult(EMPTY_LIST, 0,
+                Collections.singletonList(TERMINATING_JOB_PREFIX + SOME_CORRECT_JOB_ID_STRING));
+        mockCommandCompilation(SCANCEL, scancelCommandResult, commandArgs);
+
+        final DeleteJobFilter testDeleteJobFilter = new DeleteJobFilter(false, SOME_CORRECT_JOB_ID, passedUserName);
+        final DeletedJobInfo deletedJobInfoResult = slurmJobProvider.deleteJob(testDeleteJobFilter);
+        Assertions.assertEquals(expectedUserName, deletedJobInfoResult.getUser());
+        Assertions.assertEquals(Collections.singletonList(SOME_CORRECT_JOB_ID), deletedJobInfoResult.getId());
+    }
+
+    static Stream<Arguments> provideValidNameCasesForRequestsToDeleteJob() {
+        return Stream.of(
```

Comment:
I would also add a case with specified user but without any id

Reply:
I guess such a case is presented in the `shouldReturnCorrectListDeletedJobs` test
or maybe you meant something else?